pkg/detector/library/osv/advisory.go

```diff
@@ -0,0 +1,57 @@
+package osv
+
+import (
+	"strings"
+
+	"golang.org/x/xerrors"
+
+	dbTypes "github.com/aquasecurity/trivy-db/pkg/types"
+	"github.com/aquasecurity/trivy-db/pkg/vulnsrc/osv"
+	"github.com/aquasecurity/trivy/pkg/detector/library/comparer"
+	"github.com/aquasecurity/trivy/pkg/types"
+)
+
+// VulnSrc defines the operations on vulnerability source
+type VulnSrc interface {
+	Get(pkgName string) ([]dbTypes.Advisory, error)
+}
+
+// Advisory implements VulnSrc
+type Advisory struct {
+	vs       VulnSrc
+	comparer comparer.Comparer
+}
+
+// NewAdvisory is the factory method to return advisory
+func NewAdvisory(ecosystemName string, comparer comparer.Comparer) *Advisory {
+	return &Advisory{
+		vs:       osv.NewVulnSrc(ecosystemName),
+		comparer: comparer,
+	}
+}
+
+// DetectVulnerabilities scans package for vulnerabilities
+func (s *Advisory) DetectVulnerabilities(pkgName, pkgVer string) ([]types.DetectedVulnerability, error) {
```

Comment:
why this code can't be used instead
https://github.com/aquasecurity/trivy/blob/main/pkg/detector/library/advisory.go#L16

I think with some minimal config of keys used all this code duplication can be avoided

Reply:
changed